src/NuGet.Clients/NuGet.VisualStudio.Common/NuGetFeedbackDiagnosticFileProvider.cs

```diff
@@ -1,24 +1,110 @@
 
+using System;
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Threading.Tasks;
 using Microsoft.Internal.VisualStudio.Shell.Embeddable.Feedback;
+using Microsoft.VisualStudio.Shell;
+using NuGet.Common;
 using NuGet.PackageManagement;
+using NuGet.VisualStudio.Telemetry;
+
+#nullable enable
 
 namespace NuGet.VisualStudio.Common
 {
     [Export(typeof(IFeedbackDiagnosticFileProvider))]
     public class NuGetFeedbackDiagnosticFileProvider : IFeedbackDiagnosticFileProvider
     {
+        // Don't crash VS feedback if imports fail. Maybe broken MEF is exactly what the customer is trying to report!
+        [Import(AllowDefault = true)]
+        public ISolutionManager? SolutionManager { get; set; }
+
         public IReadOnlyCollection<string> GetFiles()
         {
-            // todo: https://github.com/NuGet/Home/issues/8605
+            string filePath = GetFilePath();
+
+            // See comments on IFeedbackDiagnosticFileProvider.GetFiles
+            ThreadHelper.JoinableTaskFactory.RunAsync(() => WriteToZip(filePath))
+                .PostOnFailure(nameof(NuGetFeedbackDiagnosticFileProvider), nameof(WriteToZip));
 
             return new List<string>()
             {
-                DependencyGraphRestoreUtility.GetDefaultDGSpecFileName()
+                filePath
             };
         }
+
+        public async Task WriteToZip(string filePath)
+        {
+            var telemetry = new TelemetryEvent("feedback");
+            telemetry["IsDebuggerAttached"] = Debugger.IsAttached;
+            bool successful = false;
+            Stopwatch sw = Stopwatch.StartNew();
+            try
+            {
+
+                using (var fileStream = File.Open(filePath, FileMode.Create, FileAccess.Write, FileShare.None))
+                using (var zip = new ZipArchive(fileStream, ZipArchiveMode.Create))
+                {
+                    await AddDgSpecAsync(zip);
+                }
+
+                successful = true;
+            }
+            finally
+            {
+                sw.Stop();
+                telemetry["successful"] = successful;
+                telemetry["duration_ms"] = sw.Elapsed.TotalMilliseconds;
+                TelemetryActivity.EmitTelemetryEvent(telemetry);
+            }
+        }
+
+        private string GetFilePath()
+        {
+            string fileNamePrefix = "nuget." + DateTime.UtcNow.ToString("yyyy-MM-dd.HH-mm-ss");
+            int attempts = 0;
+            for (; ; )
+            {
+                attempts++;
+                var fileName = attempts == 0 ? fileNamePrefix + ".zip" : fileNamePrefix + "." + attempts + ".zip";
+                var fullPath = Path.Combine(Path.GetTempPath(), fileName);
+                if (!File.Exists(fullPath))
+                {
+                    return fullPath;
+                }
+            }
```

Comment:
... then, if all attempts fail, just return a predefined, randomized fallback option

```suggestion
            }
            return  Path.Combine(Path.GetTempPath(), fileNamePrefix + "." + System.Guid.NewGuid().ToString;("D") + ".zip");
```